### PR TITLE
Add a spec for `String#encode` invalid encoding

### PR DESCRIPTION
**Pull request by ahorek**
should fail on jruby and truffleruby

### CODE REVIEW

File: core/string/encode_spec.rb [hunk 1]
```diff
@@ -60,6 +60,12 @@
 
       "\rfoo".encode(universal_newline: true).should == "\nfoo"
     end
+
+    it "replaces invalid encoding" do
+      invalid = "\xe3\x81\xa1\xE3\x81\xFF".force_encoding("UTF-8")
+      encoded = invalid.encode("UTF-16LE", invalid: :replace, replace: '?').encode("UTF-8")
```
Comment (eregon) [inline core/string/encode_spec.rb]: This is doing 2 `encode` calls on the same line, could you check the result of the first encode call as well?
That would help when fixing the implementation to pass this spec.

File: core/string/encode_spec.rb [hunk 2]
```diff
@@ -60,6 +60,12 @@
 
       "\rfoo".encode(universal_newline: true).should == "\nfoo"
     end
+
+    it "replaces invalid encoding" do
+      invalid = "\xe3\x81\xa1\xE3\x81\xFF".force_encoding("UTF-8")
```
Comment (eregon) [inline core/string/encode_spec.rb]: No need to `force_encoding`, it's the encoding for all String literals per the magic encoding comment at the top.

**Comment by eregon** [comment]
This looks good, could you address the two things I mentioned?

**Comment by ahorek** [comment]
done, thanks @eregon

**Pull request merged**